crates/bevy_ecs/hecs/src/query.rs

```diff
@@ -35,25 +35,27 @@ pub unsafe trait ReadOnlyFetch {}
 pub trait Fetch<'a>: Sized {
     /// Type of value to be fetched
     type Item;
+    /// A type to store state that is needed to `access`, `borrow` or `release`.
+    type State: Default;
```

Comment:
This `Default` bound probably isn't necessary and may not be desired. In my initial testing it made iterations faster by having the Default bound and passing in`&Default::default()` for `Fetch` implementations without state, compared to just setting the state to `()` and passing in `&()`. It really doesn't make a lot of real sense where the perf change was from, so it may not be worth confusing the code with an unnecessary trait bound.

More background in this comment: https://github.com/bevyengine/bevy/discussions/635#discussioncomment-92716

Reply:
Got rid of `State` in latest version.